src/util/psbt/map/input.rs

```diff
@@ -192,3 +238,35 @@ impl Map for Input {
 }
 
 impl_psbtmap_consensus_enc_dec_oding!(Input);
+
+fn impl_psbt_insert_hash_pair<H>(
+    map: &mut BTreeMap<H, Vec<u8>>,
+    raw_key: raw::Key,
+    raw_value: Vec<u8>,
+    hash_type: PsbtHash,
+) -> Result<(), encode::Error>
+where
+    H: hashes::Hash + Deserialize,
+{
+    if !raw_key.key.is_empty() {
```

Comment:
nit: We do not unwrap anu data here, so we may skip one level of indentation by reversing the condition and instantly returning error, without having `else` after